perception/navigator_vision/nodes/cvod/cvod.py

```diff
@@ -0,0 +1,263 @@
+        # find ray and then put into inu
+        ray = np.array(camera.projectPixelTo3dRay(point))
+        return R.dot(ray)
+
+    def find_closest_object_given_ray(self, objects, ray, ray_base, tol = 3):
+        # Compare a given bounding box with objects found by LIDAR. Find the closest object if any.
+        for o in objects:
+            distance = np.linalg.norm(np.cross(ray, mil_tools.rosmsg_to_numpy(o.pose.position) - ray_base))
+            if distance > tol:
+                continue
+            else:
+                return o
+        return None
+
+    def set_label_for_object(self, label, o):
+        # Try and set the label once its found.
+        try:
+            if label > 1:
+                label = "Totem"
+            else:
+                label = "Buoy"
+            all_objects = self.DBQuery_proxy(name="", cmd="{}={}".format(o.id, label)).objects
+        except rospy.ServiceException, e:
+            print ("Couldn't set {} to {}".format(o.id, label))
+
+if __name__ == '__main__':
+    # Current working directory
+    CWD_PATH = os.getcwd()
+
+    # Path to our checkpoint, which contains a checkpoint file containing further information for constructing network.
+    PATH_TO_CKPT = os.path.join(CWD_PATH, 'frozen_inference_graph.pb')
+
+    # Labels used for identification.
+    PATH_TO_LABELS = os.path.join(CWD_PATH, 'pcodar.pbtxt')
+
+    # Number of Classes
+    NUM_CLASSES = 6
+
+    # Load lable map
+    label_map = label_map_util.load_labelmap(PATH_TO_LABELS)
+    categories = label_map_util.convert_label_map_to_categories(label_map, max_num_classes=NUM_CLASSES, use_display_name=True)
+    category_index = label_map_util.create_category_index(categories)
+
+    # Loading pretrained weights graph
+    detection_graph = tensorflow.Graph()
+    with detection_graph.as_default():
+        od_graph_def = tensorflow.GraphDef()
+        with tensorflow.gfile.GFile(PATH_TO_CKPT, 'rb') as fid:
+            serialized_graph = fid.read()
+            od_graph_def.ParseFromString(serialized_graph)
+            tensorflow.import_graph_def(od_graph_def, name='')
+
+        sess = tensorflow.Session(graph=detection_graph)
+
+    # Input tensor is the image
+    image_tensor = detection_graph.get_tensor_by_name('image_tensor:0')
+
+    # Output tensors are the detection boxes, scores, and classes
+    # Each box represents a part of the image where a particular object was detected
+    detection_boxes = detection_graph.get_tensor_by_name('detection_boxes:0')
+
+    # Each score represents level of confidence for each of the objects.
+    # The score is shown on the result image, together with the class label.
+    detection_scores = detection_graph.get_tensor_by_name('detection_scores:0')
+    detection_classes = detection_graph.get_tensor_by_name('detection_classes:0')
+
+    # Number of objects detected
+    num_detections = detection_graph.get_tensor_by_name('num_detections:0')
```

Comment:
All of these initializations needed for the class to run should be inside the class constructor or a separate init method. Actually surprised it works because there weren't declared as globals